src/backend/distributed/planner/multi_router_planner.c

```diff
@@ -3810,7 +3812,7 @@ DeferErrorIfUnsupportedRouterPlannableSelectQuery(Query *query)
 	 * then the query will anyway happen on the coordinator, so we can
 	 * allow nextval.
 	 */
-	if (contain_nextval_expression_walker((Node *) query->targetList, NULL) &&
+	if (QueryContainsNextval(query) &&
```

Comment:
If the CTE is inlined, this check will return false for the following INSERT ... SELECT statement.

```
-- Create the table
CREATE TABLE version (
    id bigserial NOT NULL,
    description varchar(255),
    PRIMARY KEY (id)
);

-- Register the table as distributed.
SELECT create_distributed_table('version', 'id');

-- Insert an initial row.
INSERT INTO version (description) VALUES ('Version 1');

-- Use a CTE and perform an INSERT ... SELECT.
WITH v AS (
    SELECT * FROM version WHERE description = 'Version 1'
)
INSERT INTO version (description)
SELECT description
FROM v;
```